routers/api/v1/repo/issue.go

```diff
@@ -23,6 +24,131 @@ import (
 	milestone_service "code.gitea.io/gitea/services/milestone"
 )
 
+// SearchIssues searches for issues across the repositories that the user has access to
+func SearchIssues(ctx *context.APIContext) {
+	// swagger:operation GET /repos/issues/search issue issueSearchIssues
+	// ---
+	// summary: Search for issues across the repositories that the user has access to
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: state
+	//   in: query
+	//   description: whether issue is open or closed
+	//   type: string
+	// - name: labels
+	//   in: query
+	//   description: comma separated list of labels. Fetch only issues that have any of this labels. Non existent labels are discarded
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page number of requested issues
+	//   type: integer
+	// - name: q
+	//   in: query
+	//   description: search string
+	//   type: string
+	// - name: priority_repo_id
+	//   in: query
+	//   description: repository to prioritize in the results
+	//   type: integer
+	//   format: int64
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/IssueList"
+
+	// find repos user can access (for issue search)
+	repoIDs := make([]int64, 0)
+	issueCount := 0
+	for page := 1; ; page++ {
+		repos, count, err := models.SearchRepositoryByName(&models.SearchRepoOptions{
```

Comment:
thanks, I changed to return HTTP 400 if called when cross repo dependencies are disabled

Reply:
Sorry it took me so long to resume this review 😔. This PR looks good to me, except for this API function (`SearchIssues`) which handles too many options which I feel unrelated to the feature it's intended to serve. It should only search for the issues required for this PR, _especially_ if it will only work if `ALLOW_CROSS_REPOSITORY_DEPENDENCIES` is true. I understand the flexibility that parameters like `state` or `labels` can give, but they will not -currently- be used from the dependencies drop down box search. In fact, it should be named accordingly (e.g. `SearchIssuesForDependencies` or something).
Otherwise, I really like this feature. I've tested it with my repos and I think it's very useful and looks very nice. 🥇

Reply:
thanks @guillep2k! I guess what I was going for was since I needed a new api method to search issues across repos, I should follow the same pattern as the search function that already existed for searching one repo: "/repos/{owner}/{repo}/issues". I thought it would make sense to be feature parity with that endpoint. And then maybe it could be used by other use cases (like possibly this one: https://github.com/go-gitea/gitea/issues/2434). But I see your point, nothing else uses it at this point so it is debatable to make it general purpose. I can certainly strip it down if you still feel it should only match the functionality of the PR. Any additional thoughts?

Reply:
I think the additional options make sense in that context - it would be annoying to have to add them later when the inevitable complaint comes in that the API is inconsistent. (We're probably going to have to expose all of the search repos options too though - which I'm not certain that this code does.)

Reply:
OK, then the check on `ALLOW_CROSS_REPOSITORY_DEPENDENCIES` to keep the client on adding them without permission should be moved to the function that adds the dependencies (where it should have been from the start, I think).

Reply:
@guillep2k I made the change to check ALLOW_CROSS_REPOSITORY_DEPENDENCIES when adding the dependency in issue_dependency.go instead. That's much better, and thanks again for the review.